<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1055</width>
    <height>585</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TP6- Cube</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="WidgetOpenGL" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>631</width>
      <height>531</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>10</y>
      <width>171</width>
      <height>301</height>
     </rect>
    </property>
    <property name="title">
     <string>NewCube</string>
    </property>
    <widget class="QWidget" name="formLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>18</y>
       <width>151</width>
       <height>308</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>posx</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="posx">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>posy</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="posy">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>posz</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="posz">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Tangage</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="tangage">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Lacet</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="lacet">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Roulis</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="roulis">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QPushButton" name="pushButton_2">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(85, 170, 255);

border: 1px solid black;

border-radius: 2px;
</string>
        </property>
        <property name="text">
         <string>AddNewCube</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QPushButton" name="AddNewPinkCube">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 85, 255);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="text">
         <string>AddNewPinkCube</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QPushButton" name="pushButton_4">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="text">
         <string>AddNewDeAJouer</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>15</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QPushButton" name="pushButton_5">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(190, 15, 0);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <widget class="QPushButton" name="pushButton_3">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(241, 255, 76);

border: 1px solid black;

border-radius: 2px;
</string>
        </property>
        <property name="text">
         <string>AddNewLightCube</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>25</y>
      <width>151</width>
      <height>271</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="styleSheet">
        <string notr="true">background-color: rgb(170, 85, 255);

border: 1px solid black;

border-radius: 2px;
</string>
       </property>
       <property name="text">
        <string>Animation</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Field of view</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="indent">
        <number>-1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDial" name="fov">
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);

border: 1px solid black;

border-radius: 2px;
</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>108</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>659</x>
      <y>309</y>
      <width>371</width>
      <height>211</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Light Source</string>
       </property>
       <widget class="QPushButton" name="ambientButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>351</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(85, 85, 255);

border: 1px solid black;

border-radius: 2px;
</string>
        </property>
        <property name="text">
         <string>Ambient Color</string>
        </property>
       </widget>
       <widget class="QPushButton" name="diffuseButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>351</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(0, 255, 127);

border: 1px solid black;

border-radius: 2px;
</string>
        </property>
        <property name="text">
         <string>Diffuse Color</string>
        </property>
       </widget>
       <widget class="QPushButton" name="specularButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>120</y>
          <width>351</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 127);

border: 1px solid black;

border-radius: 2px;
</string>
        </property>
        <property name="text">
         <string>Specular Color</string>
        </property>
       </widget>
       <widget class="QSlider" name="SpecularCoeff">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>170</y>
          <width>321</width>
          <height>20</height>
         </rect>
        </property>
        <property name="maximum">
         <number>500</number>
        </property>
        <property name="sliderPosition">
         <number>250</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="invertedControls">
         <bool>false</bool>
        </property>
        <property name="tickInterval">
         <number>1</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>150</y>
          <width>241</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>            Specular reflexivity coefficient</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="LightX">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>41</y>
          <width>81</width>
          <height>31</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="LightY">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>40</y>
          <width>81</width>
          <height>31</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="LightZ">
        <property name="geometry">
         <rect>
          <x>230</x>
          <y>40</y>
          <width>81</width>
          <height>31</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(118, 118, 118);

border: 1px solid black;

border-radius: 2px;</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1055</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>WidgetOpenGL</class>
   <extends>QOpenGLWidget</extends>
   <header>widgetopengl.h</header>
   <slots>
    <slot>startAnimation()</slot>
    <slot>changeFov(int)</slot>
    <slot>changeOpacity(int)</slot>
    <slot>addNewCube()</slot>
    <slot>reset()</slot>
    <slot>changeAmbientLight()</slot>
    <slot>changeDiffuseLight()</slot>
    <slot>changeSpecularLight()</slot>
    <slot>changeSpecularCoefficient(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fov</sender>
   <signal>valueChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeFov(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addNewCube()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AddNewPinkCube</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addNewPinkCube()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>434</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addNewDeAJouer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>289</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>841</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ambientButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeAmbientLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>429</y>
    </hint>
    <hint type="destinationlabel">
     <x>650</x>
     <y>402</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>diffuseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeDiffuseLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>448</y>
    </hint>
    <hint type="destinationlabel">
     <x>652</x>
     <y>440</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>specularButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSpecularLight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>653</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LightX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeLightPosition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>744</x>
     <y>402</y>
    </hint>
    <hint type="destinationlabel">
     <x>703</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LightY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeLightPosition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>823</x>
     <y>391</y>
    </hint>
    <hint type="destinationlabel">
     <x>771</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LightZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeLightPosition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>1020</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SpecularCoeff</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeSpecularCoefficent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>680</x>
     <y>520</y>
    </hint>
    <hint type="destinationlabel">
     <x>656</x>
     <y>550</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addNewLightCube()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>847</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>649</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addNewCube()</slot>
  <slot>addNewPinkCube()</slot>
  <slot>addNewDeAJouer()</slot>
  <slot>reset()</slot>
  <slot>changeAmbientLight()</slot>
  <slot>changeDiffuseLight()</slot>
  <slot>changeSpecularLight()</slot>
  <slot>changeSpecularCoefficient(int)</slot>
  <slot>changeLightPosition()</slot>
  <slot>changeSpecularCoefficent()</slot>
  <slot>addNewLightCube()</slot>
 </slots>
</ui>
